main/src/cgeo/geocaching/maps/mapsforge/v6/caches/AbstractCachesOverlay.java

```
@@ -296,7 +296,11 @@ protected void clearLayers() {
         synchronized (this.bundleRef.get().getMapView()) {
             for (final GeoitemLayer layer : layerList) {
                 geoEntries.remove(new GeoEntry(layer.getItemCode(), overlayId));
-                layers.remove(layer);
+                try {
+                    layers.remove(layer);
+                } catch (IllegalStateException ignore) {
+                    // ignore IllegalStateException("layer is not assigned")
```

Maybe add Log on debug level:

`Log.d("Ignored exception on layer removal", ignore); `

---

updated PR accordingly

---

great, thanks!